### ISSUE TITLE
Training slows down with gluon and horovod on single server with 4GPUs

### ISSUE THREAD

**Issue by yiwuyao3863** [horovod/horovod#860]
When I use gluon and horovod to tain resnet101 on single server with 4GPUs, the training speed slows down after 100 steps, from 400 samples/s to 80 samples/s。The size of train-set is 4819499, the batch-size for each GPU is 60:
![image](https://user-images.githubusercontent.com/14269389/53463906-52746f80-3a83-11e9-981c-29841eb09953.png)

**Comment by alsrgv** [comment]
cc @yuxihu @apeforest

**Comment by yuxihu** [comment]
@yiwuyao3863 Please provide your information about your setup, such as training script, Horovod version, MXNet version, CUDA version, etc.

**Comment by yuxihu** [comment]
@alsrgv Can we have a template in Horovod for users to provide their environment/setup information when they asks a question or report an issue? Right now we have to go through a couple of conversations in order to get these information. Thanks.

**Comment by alsrgv** [comment]
@yuxihu, good point, adding.

**Comment by yiwuyao3863** [comment]
> @yiwuyao3863 Please provide your information about your setup, such as training script, Horovod version, MXNet version, CUDA version, etc.

Maybe the random access of disk leads to slow training, when the Shuffle=False in RecordIO, the training speed maintains high. But how can I solve this problem? Thanks~

**Comment by yiwuyao3863** [comment]
> @alsrgv Can we have a template in Horovod for users to provide their environment/setup information when they asks a question or report an issue? Right now we have to go through a couple of conversations in order to get these information. Thanks.

ENV CUDNN_VERSION=7.4.1.5-1+cuda9.0
ENV NCCL_VERSION=2.4.2-1+cuda9.0
ENV MXNET_URL=https://s3-us-west-2.amazonaws.com/mxnet-python-packages-gcc5/mxnet_cu90_gcc5-1.4.0-py2.py3-none-manylinux1_x86_64.whl

Thanks~~

**Comment by yiwuyao3863** [comment]
@yuxihu 
ENV CUDNN_VERSION=7.4.1.5-1+cuda9.0
ENV NCCL_VERSION=2.4.2-1+cuda9.0
ENV MXNET_URL=https://s3-us-west-2.amazonaws.com/mxnet-python-packages-gcc5/mxnet_cu90_gcc5-1.4.0-py2.py3-none-manylinux1_x86_64.wh
Thanks~~

**Comment by yiwuyao3863** [comment]
@alsrgv @yuxihu 
Maybe the random access of disk leads to slow training, when the Shuffle=False in RecordIO, the training speed maintains high. But how can I solve this problem? Thanks~

**Comment by yuxihu** [comment]
Usually we load the data from disk to tmpfs. But you need to have enough memory to host your training data. Here is the command I use:
```
sudo mkdir -p /media/ramdisk && sudo mount -t tmpfs -o size=150G tmpfs /media/ramdisk && cp ~/data/* /media/ramdisk/
```

**Comment by yiwuyao3863** [comment]
@yuxihu Our dataset is ~400GB, maybe the space is not enough~~
